### DISCUSSION TITLE
Where can I place a PHP file to link to?

### DISCUSSION THREAD

**Discussion by BsNoSi** [datenstrom/community#471] [Ask a question] [answered]
I want to start a php file from a link. How can I achieve this?

Is it save to place such a file in the content area?

Placing it in extensions folder leads to an error, resources doesn't have the required access rights.

**Comment by schulle4u** [comment]
I would put it somewhere in the `/media` directory or in Yellows root directory.

**Comment by BsNoSi** [comment]
Your redirect extenstion lead me to a new strategy for my [cmail extension](https://github.com/BsNoSi/yellow-extension-cmail). Pointing to a php file in `/media` hides e-mail address from e-mail grabbern scanning page code. 

Possibly best would be using a form with script but this may lead to a bunch of reqired CSS to inline it comparably to a plain `<a href=…>…</a>´ code currently used.

Is there an evidence that this code possibly opens backdoors to hack a site? I have no experience in this practice of calling php files, therefore improving or securing hints are very welcome!

**Comment by markseuffert** [comment, marked as answer]
Here are some tips. If you want to integrate software with Datenstrom Yellow, I recommend to write an extension and put your PHP files into the the `system/workers` folders. Customisable files can go into the `system/layouts` folder and `system/themes` folder. If you want to use software independently of Datenstrom Yellow, I recommend to put it into your own folder, outside of Yellow. For example create a `cgi-bin` folder and put your own PHP files there.

Don't mix both design concepts, it makes things complicated.

**Comment by BsNoSi** [comment]
I redesigned »cmail« to use given structures of YELLOW CMS instead of additional php files. Sometimes it takes some time to "get into it". Cmail now follows (I hope so…) the given design concepts.
